nifi-extension-bundles/nifi-poi-bundle/nifi-poi-services/src/main/java/org/apache/nifi/excel/StandardCellFieldTypeReader.java

```diff
@@ -0,0 +1,104 @@
+        this.timeValueInference = Objects.requireNonNull(timeValueInference, "Time Value Inference required");
+    }
+
+    /**
+     * Infer Cell Field Type and append possible Data Types to mapped Field Type Inference information
+     *
+     * @param cell Spreadsheet Cell required
+     * @param fieldName Cell field name for tracking in Field Types
+     * @param fieldTypes Map of Field Name to Field Type Inference information
+     */
+    @Override
+    public void inferCellFieldType(final Cell cell, final String fieldName, final Map<String, FieldTypeInference> fieldTypes) {
+        Objects.requireNonNull(cell, "Cell required");
+        Objects.requireNonNull(fieldName, "Field Name required");
+        Objects.requireNonNull(fieldTypes, "Field Types required");
+
+        final FieldTypeInference fieldTypeInference = fieldTypes.computeIfAbsent(fieldName, key -> new FieldTypeInference());
+        final DataType dataType = getCellDataType(cell);
+        fieldTypeInference.addPossibleDataType(dataType);
+    }
+
+    /**
+     * Get Record Data Type from Spreadsheet Cell Type and additional resolution of NUMERIC and STRING types
+     *
+     * @param cell Spreadsheet Cell required
+     * @return Record Data Type
+     */
+    @Override
+    public DataType getCellDataType(final Cell cell) {
+        Objects.requireNonNull(cell, "Cell required");
+
+        final CellType cellType = cell.getCellType();
+
+        final DataType dataType;
+
+        if (CellType.NUMERIC == cellType) {
+            // Date Formatting check limited to NUMERIC Cell Types
+            if (DateUtil.isCellDateFormatted(cell)) {
+                // Default to TIMESTAMP for date formatted values using standard Date Time Pattern
+                dataType = RecordFieldType.TIMESTAMP.getDataType();
+            } else {
+                final double numericCellValue = cell.getNumericCellValue();
+                final long roundedCellValue = (long) numericCellValue;
+                if (roundedCellValue == numericCellValue) {
+                    dataType = RecordFieldType.LONG.getDataType();
+                } else {
+                    // Default to DOUBLE for NUMERIC values following cell.getNumericCellValue()
+                    dataType = RecordFieldType.DOUBLE.getDataType();
+                }
+            }
+        } else if (CellType.BOOLEAN == cellType) {
+            dataType = RecordFieldType.BOOLEAN.getDataType();
+        } else if (CellType.STRING == cellType) {
+            final String cellValue = cell.getRichStringCellValue().getString();
+            // Attempt Time Value inference for STRING cell values
+            final Optional<DataType> timeDataType = timeValueInference.getDataType(cellValue);
+            dataType = timeDataType.orElse(RecordFieldType.STRING.getDataType());
+        } else if (CellType.FORMULA == cellType) {
+            dataType = RecordFieldType.STRING.getDataType();
+        } else {
```

Comment:
I am not sure this is correct because the formula has to be evaluated and can evaluated to a number or a boolean. The text of the formula is a string but I assume most end user's are not interested in the formula text but rather what the formula evaluates to.

Reply:
That's a good point about the evaluated value of the cell based on the formula. The current `ExcelRecordReader.getCellValue()` method calls `Cell.getStringCellValue()` for the `FORMULA` Cell Type, so this behavior aligns with that approach. This could be changed and improved, but that might make sense as a separate change. What do you think?

Reply:
In my PR I was trying to correct that logic as I had not realized `Cell.getStringCellValue()` get the text of the formula. That is why I had added to ExcelUtils

 ```
private static final DataFormatter DATA_FORMATTER = new DataFormatter(Locale.getDefault());

    static {
        DATA_FORMATTER.setUseCachedValuesForFormulaCells(true);
    }
```

as DataFormatter can obtain the formula values after calling `setUseCachedValuesForFormulaCells(true)`

Reply:
I pushed another update to improve the handling of Formula Cells, based on the Cached Formula Result Type. This follows the same general behavior of the `DataFormatter` when it is configured to use cached values from formula cells.